### PR TITLE
Allow selected pods to prevent reboots

### PR DESCRIPTION
**Pull request by awh**
This PR adds functionality to block individual nodes from rebooting in the presence of specific pods, as determined by a label selector:

```
--blocking-pod-selector=runtime=long,cost=expensive
```

see documentation in the commit for more details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,15 +68,17 @@ The following arguments can be passed to kured via the daemonset pod template:
...
-      --ds-namespace string         name of daemonset on which to place lock (default "kured")
-      --lock-annotation string      annotation in which to record locking node (default "weave.works/kured-node-lock")
-      --period duration             reboot check period (default 1h0m0s)
-      --prometheus-url string       Prometheus instance to probe for active alerts
-      --reboot-sentinel string      path to file whose existence signals need to reboot (default "/var/run/reboot-required")
-      --slack-hook-url string       slack hook URL for reboot notfications
-      --slack-username string       slack username for reboot notfications (default "kured")
+      --alert-filter-regexp regexp.Regexp   alert names to ignore when checking for active alerts
+      --blocking-pod-selector stringArray   label selector identifying pods whose presence should prevent reboots
+      --ds-name string                      name of daemonset on which to place lock (default "kured")
```
Comment (bboreham) [inline README.md]: unrelated change (correcting name vs namespace?)

File: README.md [hunk 2]
```diff
@@ -67,15 +68,17 @@ The following arguments can be passed to kured via the daemonset pod template:
...
+      --ds-namespace string                 namespace containing daemonset on which to place lock (default "kube-system")
+  -h, --help                                help for kured
```
Comment (bboreham) [inline README.md]: unrelated change; looks like wrong indent

**Comment by awh** [comment]
Good spot. It's because I updated the entire flags block by cut and pasting the output of `kured -h`...
